Episode-5/after/Blapp/app/delegates/PostsControllerTableViewDelegate.rb

```diff
@@ -4,20 +4,30 @@ def tableView(tableView, didSelectRowAtIndexPath:indexPath)
     tableView.deselectRowAtIndexPath(indexPath, animated:true)
 
     if indexPath.row == 0
-      alert = UIAlertView.alloc.initWithTitle("Post Title", message:"Enter a title for your Blapp post", delegate:self, cancelButtonTitle:"Cancel", otherButtonTitles:"Continue", nil)
-      alert.alertViewStyle = UIAlertViewStylePlainTextInput
-      alert.show
+      alert = UIAlertController.alertControllerWithTitle('Post Title',
+                                                         message: 'Enter a title for your Blapp post',
+                                                         preferredStyle: UIAlertControllerStyleAlert)
+      cancel_action = UIAlertAction.actionWithTitle('Cancel', style: UIAlertActionStyleCancel, handler: nil)
+      continue_action = UIAlertAction.actionWithTitle('Continue', style: UIAlertActionStyleDefault, handler: lambda { |_| create_new_post(alert.textFields[0].text)})
+      alert.addTextFieldWithConfigurationHandler(lambda {|_|})
+      alert.addAction(cancel_action)
+      alert.addAction(continue_action)
+      presentViewController(alert, animated: true, completion: nil)
     else
       self.navigationController.pushViewController(PostDetailController.alloc.initWithPost(Post.all[indexPath.row - 1]), animated:true)
     end
   end
 
   def tableView(tableView, accessoryButtonTappedForRowWithIndexPath:indexPath)
-    alert = UIAlertView.alloc.initWithTitle("Edit Post Title", message:"Enter a title for your Blapp post", delegate:self, cancelButtonTitle:"Cancel", otherButtonTitles:"Continue", nil)
-    alert.tag = indexPath.row - 1
-    alert.alertViewStyle = UIAlertViewStylePlainTextInput
-    alert.textFieldAtIndex(0).text = Post.all[indexPath.row - 1].title
-    alert.show
+    alert = UIAlertController.alertControllerWithTitle('Edit Post Title',
+                                                       message: 'Enter a title for your Blapp post',
+                                                       preferredStyle: UIAlertControllerStyleAlert)
+    cancel_action = UIAlertAction.actionWithTitle('Cancel', style: UIAlertActionStyleCancel, handler: nil)
+    continue_action = UIAlertAction.actionWithTitle('Continue', style: UIAlertActionStyleDefault, handler: lambda { |_| update_post(indexPath.row - 1, with_title: alert.textFields[0].text)})
+    alert.addTextFieldWithConfigurationHandler(lambda {|text_field| text_field.text = Post.all[indexPath.row - 1].title})
+    alert.addAction(cancel_action)
+    alert.addAction(continue_action)
+    presentViewController(alert, animated: true, completion: nil)
```

Comment:
Chunk 2.  These are almost duplicate chunks of code.  See if you can figure out how to tease them out into a method that can be called in both places, like the method you deleted above.